common/customizable_schema.go

```diff
@@ -13,6 +15,64 @@ type CustomizableSchema struct {
 	Schema         *schema.Schema
 	path           []string
 	isSuppressDiff bool
+	context        SchemaPathContext
+}
+
+type SchemaPathContext struct {
+	// Path is used for refernces from resourceProviderRegistry as prefix, follows the format of `[a,0,b]`
+	path       []string
+	schemaPath []*schema.Schema
```

Comment:
This is risky because it is easy for developers to incorrectly set this state. I would rather introduce a new pair struct and have a list of pairs here.